contracts/JoepegAuctionHouse.sol

```diff
@@ -0,0 +1,960 @@
+    /// @dev (collection address => token id => english auction)
+    mapping(address => mapping(uint256 => EnglishAuction))
+        public englishAuctions;
+
+    /// @notice Required minimum percent increase from last bid in order to
+    /// place a new bid on an English Auction
+    uint256 public englishAuctionMinBidIncrementPct;
+
+    /// @notice Represents both:
+    /// - Number of seconds before an English Auction ends where any new
+    ///   bid will extend the auction's end time
+    /// - Number of seconds to extend an English Auction's end time by
+    uint256 public englishAuctionRefreshTime;
+
+    event DutchAuctionStart(
+        address indexed creator,
+        address currency,
+        address indexed collection,
+        uint256 indexed tokenId,
+        uint256 startPrice,
+        uint256 endPrice,
+        uint256 startTime,
+        uint256 endTime,
+        uint256 dropInterval
+    );
+    event DutchAuctionSettle(
+        address indexed creator,
+        address indexed buyer,
+        address indexed currency,
+        address collection,
+        uint256 tokenId,
+        uint256 price
+    );
+    event DutchAuctionCancel(
+        address indexed creator,
+        address indexed collection,
+        uint256 indexed tokenId
+    );
+
+    event EnglishAuctionStart(
+        address indexed creator,
+        address currency,
+        address indexed collection,
+        uint256 indexed tokenId,
+        uint256 startPrice,
+        uint256 startTime,
+        uint256 endTime
+    );
+    event EnglishAuctionPlaceBid(
+        address indexed creator,
+        address indexed bidder,
+        address indexed currency,
+        address collection,
+        uint256 tokenId,
+        uint256 bidAmount,
+        uint256 endTime
+    );
+    event EnglishAuctionSettle(
+        address indexed creator,
+        address indexed bidder,
+        address indexed currency,
+        address collection,
+        uint256 tokenId,
+        uint256 price
+    );
+    event EnglishAuctionCancel(
+        address indexed creator,
+        address indexed collection,
+        uint256 indexed tokenId
+    );
+
+    event NewCurrencyManager(address indexed currencyManager);
+    event NewEnglishAuctionMinBidIncrementPct(
+        uint256 englishAuctionMinBidIncrementPct
+    );
+    event NewEnglishAuctionRefreshTime(uint256 englishAuctionRefreshTime);
+    event NewProtocolFeeManager(address indexed protocolFeeManager);
+    event NewProtocolFeeRecipient(address indexed protocolFeeRecipient);
+    event NewRoyaltyFeeManager(address indexed royaltyFeeManager);
+
+    event RoyaltyPayment(
+        address indexed collection,
+        uint256 indexed tokenId,
+        address indexed royaltyRecipient,
+        address currency,
+        uint256 amount
+    );
+
+    modifier isSupportedCurrency(address _currency) {
+        if (!currencyManager.isCurrencyWhitelisted(_currency)) {
```

Comment:
yeah `WAVAX` is allowlisted inside `currencyManager` already. I think I prefer to keep logic like this in case there is  a scenario where we want to turn it off for some reason and also just to keep the source of truth to be `CurrencyManager`

Reply:
Currently the specific AVAX/WAVAX functions doesn't check the currencyManager, so there is no way to turn wavax/avax off

Reply:
My bad I mixed start and settle functions

Reply:
My bad I mixed start and settle functions